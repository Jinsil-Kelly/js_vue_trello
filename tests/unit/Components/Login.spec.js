import Login from "@/views/Login.vue";
import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);
const routes = [
  {
    path: '/login?rPath="%2f"',
    component: Login
  }
];
const router = new VueRouter({
  routes
});
describe("LoginComponent", () => {
  const build = () => {
    const options = { localVue, router };
    const wrapper = shallowMount(Login, options);
    const wrapperMounted = mount(Login, options);

    return {
      wrapper,
      wrapperMounted,
      emailInput: () => wrapper.find("#email"),
      pwdInput: () => wrapper.find("#password"),
      emailInputMounted: () => wrapperMounted.find("#email"),
      pwdInputMounted: () => wrapperMounted.find("#password"),
      button: () => wrapperMounted.find("button")
    };
  };

  it("renders the component", () => {
    // arrange
    const { wrapper } = build();

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    // arrange
    const { wrapper, emailInput, pwdInput, button } = build();
    // assert
    expect(wrapper.findAll("input").length).toEqual(2);
    expect(emailInput().exists()).toBe(true);
    expect(pwdInput().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });

  it('calls "onSubmit" func when submitting form', () => {
    // arrange
    const { wrapper, button, emailInputMounted, pwdInputMounted } = build();
    const loginForm = wrapper.find("#loginForm");
    const email = "test@test.com";
    const pwd = "123123";
    const onSubmitMock = jest.fn();
    wrapper.setMethods({ onSubmit: onSubmitMock });

    emailInputMounted().element.value = email;
    pwdInputMounted().element.value = pwd;

    // act
    emailInputMounted().trigger("input");
    pwdInputMounted().trigger("input");

    button().trigger("click");
    button().trigger("submit");
    loginForm.trigger("submit.prevent");

    // assert
    expect(onSubmitMock).toBeCalled();
  });
});
