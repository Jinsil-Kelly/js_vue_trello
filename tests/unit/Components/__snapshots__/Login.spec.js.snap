// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent renders the component 1`] = `
<div class="login">
  <h2>Log in to Trello</h2>
  <form id="loginForm">
    <div><label for="email">Email</label> <input id="email" type="text" name="email" autofocus="autofocus" placeholder="e.g., test@test.com" class="form-control"></div>
    <div><label for="password">Password</label> <input id="password" type="password" placeholder="123123" class="form-control"></div> <button type="submit" disabled="disabled" class="btn">
      Log In
    </button>
  </form>
  <!---->
</div>
`;
